File: 5295. Find N Unique Integers Sum up to Zero/5295.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sumZero(int n, int* returnSize){
    *returnSize = n;
    int* res = (int*)malloc(sizeof(int)*n);

    if (n%2 == 1){
        n--;
        *(res+n) = 0;
    }

    while(n != 0){
        *(res+n-1) = n;
        *(res+n-2) = -n;
        n -= 2;
    }

    return res;
}

File: 5295. Find N Unique Integers Sum up to Zero/_5295.py
class Solution:
    def sumZero(self, n: int) :
        res = list()
        if n%2 == 1:
            n -= 1
            res.append(0)
        
        while n != 0:
            res.append(n)
            res.append(-n)
            n -= 2

        return res 